spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ReactiveKafkaProperties.java

```diff
@@ -0,0 +1,341 @@
+		 * operation is attempted when the number of acknowledged uncommitted offsets
+		 * reaches this batch size.
+		 */
+		private int commitBatchSize;
+
+		/**
+		 * Configures commit ahead size per partition for at-most-once delivery. Before
+		 * dispatching each record, an offset ahead by this size may be committed. The
+		 * maximum number of records that may be lost if the application fails is
+		 * <code>commitAheadSize + 1</code>.
+		 */
+		private int atmostOnceCommitAheadSize;
+
+		/**
+		 * Configures the maximum number of consecutive non-fatal
+		 * {@link RetriableCommitFailedException} commit failures that are tolerated. For
+		 * manual commits, failure in commit after the configured number of attempts fails
+		 * the commit operation. For auto commits, the received Flux is terminated if the
+		 * commit does not succeed after these attempts.
+		 */
+		private int maxCommitAttempts;
+
+		/**
+		 * Set to greater than 0 to enable out of order commit sequencing. If the number
+		 * of deferred commits exceeds this value, the consumer is paused until the
+		 * deferred commits are reduced.
+		 */
+		private int maxDeferredCommits;
+
+		public Duration getPollTimeout() {
+			return this.pollTimeout;
+		}
+
+		public void setPollTimeout(Duration pollTimeout) {
+			this.pollTimeout = pollTimeout;
+		}
+
+		public Duration getCloseTimeout() {
+			return this.closeTimeout;
+		}
+
+		public void setCloseTimeout(Duration closeTimeout) {
+			this.closeTimeout = closeTimeout;
+		}
+
+		public Collection<String> getSubscribeTopics() {
+			return this.subscribeTopics;
+		}
+
+		public void setSubscribeTopics(Collection<String> subscribeTopics) {
+			this.subscribeTopics = subscribeTopics;
+		}
+
+		public Pattern getSubscribePattern() {
+			return this.subscribePattern;
+		}
+
+		public void setSubscribePattern(Pattern subscribePattern) {
+			this.subscribePattern = subscribePattern;
+		}
+
+		public Duration getCommitInterval() {
+			return this.commitInterval;
+		}
+
+		public void setCommitInterval(Duration commitInterval) {
+			this.commitInterval = commitInterval;
+		}
+
+		public int getCommitBatchSize() {
+			return this.commitBatchSize;
+		}
+
+		public void setCommitBatchSize(int commitBatchSize) {
+			this.commitBatchSize = commitBatchSize;
+		}
+
+		public int getAtmostOnceCommitAheadSize() {
+			return this.atmostOnceCommitAheadSize;
+		}
+
+		public void setAtmostOnceCommitAheadSize(int atmostOnceCommitAheadSize) {
+			this.atmostOnceCommitAheadSize = atmostOnceCommitAheadSize;
+		}
+
+		public int getMaxCommitAttempts() {
+			return this.maxCommitAttempts;
+		}
+
+		public void setMaxCommitAttempts(int maxCommitAttempts) {
+			this.maxCommitAttempts = maxCommitAttempts;
+		}
+
+		public int getMaxDeferredCommits() {
+			return this.maxDeferredCommits;
+		}
+
+		public void setMaxDeferredCommits(int maxDeferredCommits) {
+			this.maxDeferredCommits = maxDeferredCommits;
+		}
+
+		public Map<String, Object> buildProperties() {
```

Comment:
We don't need any of these `build*Properties()` methods; all the kafka specific properties are already handled by parts of `KafkaProperties`.